components/p3a/brave_p3a_service.cc

```diff
@@ -194,8 +194,10 @@ base::TimeDelta TimeDeltaTillMonday(base::Time time) {
     days_till_monday = 1;
   }
 
-  base::TimeDelta result = base::TimeDelta::FromDays(days_till_monday) -
-                           (time - time.LocalMidnight());
+  // Adding few hours of padding to prevent potential problems with DST.
+  base::Time result =
+      (time.LocalMidnight() + base::Days(days_till_monday) + base::Hours(4))
+          .LocalMidnight();
```

Comment:
The check in `UpdateRotationTimer` will catch this, but I'm still worried that we can return a negative value here. Hopefully whatever memory corruption causes that would take down the browser before we repeat the DST spin-of-death, but I'd feel better if we clamped `days_till_monday >= 1` here, or `DCHECK(result > base::Time::Now())`.